python/lsst/ts/planning/tool/zephyr_interface.py

```diff
@@ -397,30 +445,32 @@ async def get_test_execution(self, test_execution_key, parse="raw"):
             if test_execution[user] and parse == "simple":
                 test_execution[user] = test_execution[user]["displayName"]
 
-        return test_execution
-
-    async def get_test_execution_steps(self, test_execution_key):
-        """
-        Get all the steps in a test execution.
-
-        Parameters
-        ----------
-        test_execution_key : str
-            The key of the test execution.
-
-        Returns
-        -------
-        dict
-            A dictionary containing the steps of the test execution.
+        # Fill the payload with the test execution steps
+        if complement:
+            # Get the test execution and test case steps, they should have the
+            #   same number of elements (steps).
+            test_execution_steps = await self.get_steps(
+                test_execution_key, complement=complement
+            )
+            test_case_steps = await self.get_steps(
+                test_execution["testCase"]["key"], complement=complement
+            )
```

Comment:
I'm getting errors on this line. You are doing `test_execution["testCase"]["key"]`, but `test_execution["testCase"]` is not a dict and it already has the test case key.
```suggestion
            test_case_steps = await self.get_steps(
                test_execution["testCase"], complement=complement
            )
```